src/DynamoCoreWpf/ViewModels/Watch3D/DefaultWatch3DViewModel.cs

```diff
@@ -594,12 +597,31 @@ protected virtual void SelectionChangedHandler(object sender, NotifyCollectionCh
             // Override in derived classes
         }
 
-        protected virtual void OnRenderPackagesUpdated(NodeModel node, IEnumerable<IRenderPackage> packages)
+        protected virtual void OnRenderPackagesUpdated(NodeModel node, RenderPackageCache packages)
         {
             RemoveGeometryForNode(node);
-            if(packages.Any())
+            if (packages.IsEmpty())
+                return;
+
+            // If no attached model update for all render packages
+            if (watchModel == null)
             {
                 AddGeometryForRenderPackages(packages);
+                return;
+            }
+
+            // Only update for render packages from the connected output port
+            var inputId = watchModel.InPorts[0].Connectors[0].Start.GUID;
```

Comment:
seems like we should check for null here, what if connectors is somehow empty - or ports?

Reply:
I can add null checks for each of these indirection layers. If connectors can be empty it seems to follow that any of the other items could cause an invalid state as well.